### DISCUSSION TITLE
participants.map is not a function

### DISCUSSION THREAD

**Discussion by d-opri** [neuefische/questions#478] [Web] [answered]
moin moin,

i have a component that is supposed to render new "participant" list items on button click, for that i have my useState, handleAdd and onChange functions. now in the return statement of my component i want to map over "participants" to create new list items, but i get the error "participants.map is not a function"
ive had this before but i cannot remember how to resolve it. 

😿 its probably something tiny but my brains fried

**Comment by JoergWolff** [comment]
Hi d-opri(Danie), how about a ```console.log(participants)```near by. If the result is undefined, you got this message.

**Comment by doemser** [comment, marked as answer]
If map is "not a function", the data type you want to map over is usually not an array, or not yet if you are fetching.

**Comment by ThomasF85** [comment]
Hi Danie,

the error occurs because your participant state is not an array (it might be just a single object) so it does not have the map method.
Please provide the code and screenshots of the error message. Otherwise it is impossible to tell where the root of the problem is.

**Reply by d-opri** [reply]
morning, thanks for all the help! my code loooks like this

```
  const [participants, setParticipants] = useState([
    {
      name: "",
      id: uuid(),
      balance: 0,
    },
  ]);
```
and then later on in the return statement

```
        {participants?.map((participant) => (
            /.../
        ))}
```

<img width="587" alt="Screenshot 2023-02-01 at 10 04 19" src="https://user-images.githubusercontent.com/118123298/215999253-23b32ebc-5a88-44ce-a90c-fa655c26b1ed.png">

**Reply by doemser** [reply]
Please provide a current PR, so that we can take a look at it in context.

**Reply by d-opri** [reply]
i actually managed to find my mistakes, one was in the handlechange function and then i also forgor to pass the id to the map function.. so it works now!

thanks for the support
